cmd/mp4info/mp4info.go

```diff
@@ -37,44 +37,71 @@ func main() {
-		fmt.Printf("    duration:       %d (media timescale units)\n", trak.Mdia.Mdhd.Duration)
-		fmt.Printf("    duration:       %02.0f (ms)\n", math.Floor(float64(trak.Mdia.Mdhd.Duration)/float64(trak.Mdia.Mdhd.Timescale)*1000))
-
-		if trak.Tkhd.GetWidth() != 0 || trak.Tkhd.GetHeight() != 0 {
-			fmt.Printf("  display width:         %d\n", trak.Tkhd.GetWidth())
-			fmt.Printf("  display height:        %d\n", trak.Tkhd.GetHeight())
+	if (f.Moov != nil) {
+		fmt.Println("Movie:")
+
+		fmt.Printf("  duration: %d ms / %d (%s)\n",
+			f.Moov.Mvhd.Duration, f.Moov.Mvhd.Timescale,
+			atom.GetDurationString(f.Moov.Mvhd.Duration, f.Moov.Mvhd.Timescale))
+		fmt.Printf("  fragments:    %t\n", f.IsFragmented)
+		fmt.Printf("  timescale:    %d\n\n", f.Moov.Mvhd.Timescale)
+	
+		fmt.Printf("Found %d Tracks\n\n", len(f.Moov.Traks))
+	
+		for _, trak := range f.Moov.Traks {
+			fmt.Printf("Track %d:\n", trak.Tkhd.TrackID)
+			fmt.Printf("  flags:    %d %s\n", trak.Tkhd.Flags, getFlags(trak.Tkhd.Flags))
+			fmt.Printf("  id:       %d\n", trak.Tkhd.TrackID)
+			fmt.Printf("  type:     %s\n", getHandlerType(trak.Mdia.Hdlr.Handler))
+			fmt.Printf("  duration: %d ms\n", trak.Tkhd.Duration)
+			fmt.Printf("  language: %s\n", trak.Mdia.Mdhd.LanguageString)
+			fmt.Printf("  width:    %d\n", trak.Tkhd.Width/(1<<16))
+			fmt.Printf("  height:   %d\n", trak.Tkhd.Height/(1<<16))
+	
+			fmt.Println("  media:")
+			fmt.Printf("    sample count:   %d\n", trak.Mdia.Minf.Stbl.Stts.SampleCounts[0])
+			fmt.Printf("    timescale:      %d\n", trak.Mdia.Mdhd.Timescale)
+			fmt.Printf("    duration:       %d (media timescale units)\n", trak.Mdia.Mdhd.Duration)
+			fmt.Printf("    duration:       %02.0f (ms)\n", math.Floor(float64(trak.Mdia.Mdhd.Duration)/float64(trak.Mdia.Mdhd.Timescale)*1000))
+	
+			if trak.Tkhd.GetWidth() != 0 || trak.Tkhd.GetHeight() != 0 {
+				fmt.Printf("  display width:         %d\n", trak.Tkhd.GetWidth())
+				fmt.Printf("  display height:        %d\n", trak.Tkhd.GetHeight())
+			}
+	
+			if getHandlerType(trak.Mdia.Hdlr.Handler) == "Video" {
+				sampleCounts := 1000 * trak.Mdia.Minf.Stbl.Stts.SampleCounts[0]
+				durationMs := math.Floor(float64(trak.Mdia.Mdhd.Duration) / float64(trak.Mdia.Mdhd.Timescale) * 1000)
+				fmt.Printf("  frame rate (computed): %.2f\n", float64(sampleCounts)/durationMs)
+			}
+			fmt.Println("")
 		}
-
-		if getHandlerType(trak.Mdia.Hdlr.Handler) == "Video" {
-			sampleCounts := 1000 * trak.Mdia.Minf.Stbl.Stts.SampleCounts[0]
-			durationMs := math.Floor(float64(trak.Mdia.Mdhd.Duration) / float64(trak.Mdia.Mdhd.Timescale) * 1000)
-			fmt.Printf("  frame rate (computed): %.2f\n", float64(sampleCounts)/durationMs)
+	} else {
+		fmt.Printf("MOOV atom not found...\n\nDumping all boxes found:\n\n")
+	
+		info, err := f.Stat()
+		if err != nil {
+			fmt.Fprintln(os.Stderr, err)
+			return
+		}
+	
+		f.Size = info.Size()
+	
+		boxes := atom.ReadBoxes(f, int64(0), f.Size)
```

Comment:
I haven't tried running this fork, since I don't have an mp4 without a `moov` box, but it looks like the boxes are being re-parsed again here, except directly. Is it possible to traverse the boxes from `box.File` using the contents from `f, err := mp4.Open(input)` above?

Reply:
You're right — they _are_ being re-parsed again! (I told you, it was a quick & dirty hack...)

Reply:
Hey, no worries! Thanks for the effort anyhow!